packages/block-library/src/media-text/edit.native.js

```diff
@@ -259,23 +311,60 @@ class MediaTextEdit extends Component {
 			setAttributes( { verticalAlignment: alignment } );
 		};
 
+		const controls = (
+			<InspectorControls>
+				<PanelBody title={ __( 'Media & Text settings' ) }>
+					<ToggleControl
+						label={ __( 'Crop image to fill entire column' ) }
+						checked={ imageFill }
+						onChange={ () =>
+							setAttributes( {
+								imageFill: ! imageFill,
+							} )
+						}
+					/>
+				</PanelBody>
+			</InspectorControls>
+		);
+
 		return (
 			<>
+				{ mediaType === MEDIA_TYPE_IMAGE && controls }
 				<BlockControls>
-					<ToolbarGroup controls={ toolbarControls } />
-					<BlockVerticalAlignmentToolbar
-						onChange={ onVerticalAlignmentChange }
-						value={ verticalAlignment }
-					/>
+					{ ( isMediaSelected || mediaType === MEDIA_TYPE_VIDEO ) && (
+						<ToolbarGroup>
+							<Button
+								label={ __( 'Edit media' ) }
+								icon={ replace }
+								onClick={ this.onReplaceMedia }
+							/>
+						</ToolbarGroup>
+					) }
+					{ ( ! isMediaSelected ||
+						mediaType === MEDIA_TYPE_VIDEO ) && (
+						<>
+							<ToolbarGroup controls={ toolbarControls } />
+							<BlockVerticalAlignmentToolbar
+								onChange={ onVerticalAlignmentChange }
+								value={ verticalAlignment }
+							/>
+						</>
+					) }
 				</BlockControls>
 				<View
 					style={ containerStyles }
 					onLayout={ this.onLayoutChange }
 				>
 					<View
-						style={ { width: widthString, ...mediaContainerStyle } }
+						style={ [
+							{ flex: 1 },
```

Comment:
Can you add the `{ flex: 1 }` to the `mediaContainerStyle`?

Reply:
Nice! Updated 👍